spec/controllers/orders_controller_spec.rb

```diff
@@ -69,97 +53,368 @@
     it "assigns @order" do
       get :checkout
 
-      expect(assigns(:order)).to eq(Order.find(session[:order_id]))
-    end
-
-    # it "assigns @order_items" do
-    #   get :checkout
-    #
-    #   expect(assigns(:order_items)).to eq(OrderItem.all)
-    #   expect(assigns(:order_items_count)).to eq(@order_items_count)
-    # end
-
-    # it "redirects to receipt page if order status is not pending" do
-    #   Order.find(session[:order_id]).update(status: "paid")
-    #   Order.find(session[:order_id]).save
-    #   Order.find(session[:order_id]).reload
-    #
-    #   get :checkout
-    #
-    #   expect(response).to have_http_status(302)
-    #   expect(response).to redirect_to(receipt_path)
-    # end
+      expect(assigns(:order)).to eq(test_order)
+    end
+
+    context "adjusting order item quantities" do
+      before :each do
+        @order = Order.create
+        @product = Product.create(name: "34234ujoiujhe", stock: 1, price: 1, seller_id: 1)
+        @item = OrderItem.create(product_id: @product.id, order_id: @order.id, quantity_ordered: 5)
+        session[:order_id] = @order.id
+      end
+
+      it "adjusts order item quantities as needed to match available stock" do
+        get :checkout
+        @item.reload
+
+        expect(@item.quantity_ordered).to eq(1)
+      end
+
+      it "flashes an error message so the user will understand altered values" do
+        get :checkout
+        @item.reload
+
+        expect(flash[:errors].keys).to include(:product_stock)
+      end
+    end
   end
 
-  describe "GET receipt" do
+  describe "POST #add_to_cart" do
     before :each do
-      @order_items_count = number_of_cart_items_wanted = 10
-      number_of_cart_items_wanted.times do
-        OrderItem.create(order_id: 1, product_id: (1..10).to_a.sample, quantity_ordered: 5)
+      session[:order_id] = test_order.id
+    end
+
+    it "assigns @order" do
```

Comment:
Here is another set of tests where it would be preferred to describe the behaviors rather than what the specific code is doing